tiny-menu.el

```diff
@@ -37,22 +37,43 @@
   "The menu heading shown in the selection menu for `tiny-menu'."
   :group 'tiny-menu)
 
+(defvar tiny-menu-forever nil
+  "If menu transitions are omitted, stay within the same menu until quit")
+
 (defvar tiny-menu-items
   '(())
   "An alist of menus.
 
 The keys in the alist are simple strings used to reference the menu in
-calls to `tiny-menu' and the values are lists with three elements:
-A raw character to use as the selection key, such as `?a'; a string to
-use in the menu display, and a function to call when that item is
-selected.
+calls to `tiny-menu' and the values are lists with four elements:
+a raw character to use as the selection key (such as `?a'), a string to
+use in the menu display, a function to call when that item is
+selected, and a new menu to display once the function has been run.
+
+If the new menu is omitted, and `tiny-menu-forever' is non-nil,
+then the default is to remain in the current menu. If
+`tiny-menu-forever' is nil, then omitting a new menu results in
+tiny-menu terminating after execution. The special value of
+\"quit\" indicates an unconditional quit from tiny-menu, and the
+value \"root\", and any invalid menu name, indicates an
+unconditional transition to the menu of menus.
 
 The data structure should look like:
 
-'((\"menu-1\" (?a \"First item\" function-to-call-for-item-1)
-            (?b \"Second item\" function-to-call-for-item-2))
-  (\"menu-2\" (?z \"First item\" function-to-call-for-item-1)
-            (?x \"Second item\" function-to-call-for-item-2)))")
+'((\"menu-name-1\" (\"menu-display-name\" (?a \"First item\" function-to-call-for-item-1 \"transition-menu\")
+      (?b \"Second item\" function-to-call-for-item-2 \"transition-menu\")))
+((\"menu-name-2 (\"\"menu-display-name\" (?z \"First item\" function-to-call-for-item-1 \"transition-menu\")
+                 (?x \"Second item\" function-to-call-for-item-2 \"transition-menu\")))))")
+
+(defun tiny-menu--lookup-transition (current-menu next-menu)
+  (if (null next-menu)
+      (when tiny-menu-forever current-menu)
+    (cond
+     ((string-equal "quit" next-menu) nil)
+     ((string-equal "root" next-menu) (tiny-menu--menu-of-menus))
+     (t (if (assoc next-menu tiny-menu-items)
+            (cadr (assoc next-menu tiny-menu-items))
+          (tiny-menu--menu-of-menus))))))
```

Comment:
If the next menu specified doesn't exist, would it make more sense to print an error message and quit? Personally I would output a message like `The next menu "menu name" is not a valid menu.`